sdk/include/hipdnn_sdk/test_utilities/cpu_fp_reference_validation.hpp

```diff
@@ -0,0 +1,71 @@
+// Copyright © Advanced Micro Devices, Inc., or its affiliates.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <hipdnn_sdk/test_utilities/reference_validation_interface.hpp>
+
+namespace hipdnn_sdk
+{
+namespace reference_test_utilities
+{
+
+using namespace hipdnn_sdk::utilities;
+
+template <class T>
+class Cpu_fp_reference_validation : public Reference_validation_interface
+{
+public:
+    Cpu_fp_reference_validation(T absolute_tolerance = std::numeric_limits<T>::epsilon(),
+                                T relative_tolerance = std::numeric_limits<T>::epsilon())
+        : _absolute_tolerance(absolute_tolerance)
+        , _relative_tolerance(relative_tolerance)
+    {
+        if(absolute_tolerance < static_cast<T>(0.0f) || relative_tolerance < static_cast<T>(0.0f))
+        {
+            throw std::invalid_argument("Tolerances must be non-negative");
+        }
+    }
+
+    ~Cpu_fp_reference_validation() override = default;
+
+    bool compare_buffers(const Migratable_memory& reference,
+                         const Migratable_memory& implementation) override
+    {
+        if(reference.count() != implementation.count())
+        {
+            return false;
+        }
+
+        size_t element_count = reference.count();
+        const T* ref_data = reference.host_data<T>();
+        const T* impl_data = implementation.host_data<T>();
+
```

Comment:
We can also check if cpu and gpu buffers are all zero or inf, if so return false.